modules/masterbar/masterbar.php

```diff
@@ -0,0 +1,821 @@
+<?php
+
+require_once dirname( __FILE__ ) . '/rtl-admin-bar.php';
+
+class A8C_WPCOM_Masterbar {
+	private $locale;
+
+	private $user_id;
+	private $user_data;
+	private $user_login;
+	private $display_name;
+	private $primary_site_slug;
+	private $user_text_direction;
+
+	function __construct() {
+		$this->locale  = $this->get_locale();
+		$this->user_id = get_current_user_id();
+
+		// Limit the masterbar to be shown only to connected Jetpack users.
+		if ( ! Jetpack::is_user_connected( $this->user_id ) ) {
+			return;
+		}
+
+		$this->user_data = Jetpack::get_connected_user_data( $this->user_id );
+		$this->user_login = $this->user_data['login'];
+		$this->display_name = $this->user_data['display_name'];
+		$this->primary_site_slug = $this->get_site_slug();
+		// We need to use user's setting here, instead of relying on current blog's text direction
+		$this->user_text_direction = $this->user_data['text_direction'];
+
+		if ( $this->is_rtl() ) {
+			// Extend core WP_Admin_Bar class in order to add rtl styles
+			add_filter( 'wp_admin_bar_class', function () {
```

Comment:
Could you remove this anonymous function, as this will create Fatal errors in PHP 5.2?

Thanks!